<launch>
  <!-- ******************************* Arguments ******************************* -->
  <arg name="concert_name" default="concert_master"/>
  <arg name="concert_hub_uri" default="http://localhost:6380"/>
  <arg name="concert_auto_invite" default="true"/>
  <arg name="gateway_watch_loop_period" default="2"/> <!-- Polling period for multimaster advertising/flipping -->

  <!-- ********************************* Hub *********************************** -->
  <include file="$(find rocon_hub)/launch/hub.launch">
    <arg name="hub_name" value="$(arg concert_name)" />
    <arg name="hub_port" value="6380" />
  </include>
  
  <!-- ******************************* Zeroconf ******************************** -->
  <node ns="zeroconf" pkg="zeroconf_avahi" type="zeroconf" name="zeroconf"/>

  <!-- ******************************** Gateway ******************************** -->
  <node pkg="rocon_gateway" type="gateway.py" name="gateway">
    <rosparam command="load" file="$(find rocon_gateway)/param/default.yaml"/>
    <rosparam command="load" file="$(find rocon_gateway)/param/default_blacklist.yaml"/>
    <param name="hub_uri" value="$(arg concert_hub_uri)"/>
    <param name="name" value="$(arg concert_name)"/>
    <param name="firewall" value="false"/>
    <param name="watch_loop_period" value="$(arg gateway_watch_loop_period)"/>
    <param name="hub_whitelist" value="$(arg concert_hub_uri)"/>
  </node>

  <!-- ******************************* Conductor ******************************* -->
  <node pkg="concert_conductor" type="conductor.py" name="conductor" output="screen">
    <param name="auto_invite" value="$(arg concert_auto_invite)"/>
    <remap from="conductor/gateway_info" to="gateway/gateway_info"/>
    <remap from="conductor/remote_gateway_info" to="gateway/remote_gateway_info"/>
    <remap from="conductor/pull" to="gateway/pull"/>
    <remap from="conductor/flip" to="gateway/flip"/>
  </node>
</launch>
